relay_client/src/websocket.rs

```diff
@@ -240,15 +239,25 @@ impl Client {
     pub fn batch_subscribe_blocking(
         &self,
         topics: impl Into<Vec<Topic>>,
-    ) -> ResponseFuture<BatchSubscribe> {
-        let (request, response) = create_request(BatchSubscribe {
+    ) -> impl Future<
+        Output = Result<
```

Comment:
Why isn't the function just `async`?

Reply:
Making the function `async` would capture `self` reference for the lifetime of that future, which I wanted to avoid initially. Now _why_ would I want to avoid capturing `self` is another question. Maybe in the initial implementation there were some methods that required `&mut self`, so I didn't want to mix immutable and mutable borrowing, not sure. But yeah, now that the client API is stable-ish, we could just convert these methods to be `async` and return anonymous futures.